src/hooks/Guilds/ether-swr/guild/useGuildRegistry.ts

```diff
@@ -0,0 +1,19 @@
+import useEtherSWR from '../useEtherSWR';
+import GuildRegistry from 'contracts/GuildsRegistry.json';
+
+export const useGuildRegistry = (contractAddress: string) => {
+  const { data: totalGuilds, error } = useEtherSWR(
+    contractAddress
+      ? [contractAddress, 'getGuildsAddresses'] // get all the guilds addresses
+      : [],
+    {
+      ABIs: new Map([[contractAddress, GuildRegistry.abi]]),
+      refreshInterval: 0,
+    }
+  );
+
+  return {
+    totalGuilds,
+    error,
+  };

```

Comment:
I prefer to always return `{ data, loading, error }` and keep it consistent between hooks. By following this pattern we know that a data fetching hook will always have these props to be used by the component calling it.

The component can always rename these props if there are multiple hooks:
```js
function GuildsList({ contractAddress }) {
  // with deconstruction
  const { data: totalGuilds, loading: loadingGuilds, error: errorGuilds } = useGuildRegistry(contractAddress) 
  // or as object
  const otherHook = useAnotherHook() // otherHook.data | loading | error
```